lib/module.ts

```diff
@@ -86,21 +112,16 @@ export class Module<
   }
 
   private registerOnDataChangeHooks(state: UnwrapNestedRefs<S>): void {
-    const onDataChangeHooks = this.options.plugins
+    const onDataChangeHooks = this.plugins
       ?.filter((plugin) => plugin.onDataChange !== undefined)
       .map((plugin) => plugin.onDataChange)
     if (onDataChangeHooks?.length) {
-      watch(
-        state,
-        (value, oldValue, onCleanup) => {
-          onDataChangeHooks.forEach((callback) =>
-            callback?.(value, oldValue as NonNullable<typeof oldValue>, onCleanup),
-          )
-        },
-        {
-          immediate: true,
-        },
-      )
+      watch(state, (value, oldValue, onCleanup) => {
+        const rawState = toRaw(value)
```

Comment:
I think this is necessary to avoid an infinite watch loop, at least I couldn't find another way to allow the plugin callback to modify state. It might not be an important use case, but I think it doesn't hurt.